Add new concept `mapsets` and concept exercise `gotta-snatch-em-all`

jiegillet opened the pull request:
This PR add new concept and concept exercise, both forked from the Elm track.

Full disclosure, I used AI (via Cursor) to help translate from one language to the other, and event though I combed through everything and edited a lot of things, it was a huge time saver.

There might be stylistic differences between tracks that I'm blind to, I'm happy to change anything.

The CI is expected to fail until the `about.md` is finalized, then I'll copy it to `introduction.md` (let me know if you think some parts should be cut down) and use the generator to add the intro file to the exercise.

angelikatyborska reviewed (commented):
Excellent exercise, very good `about.me`, I had no trouble solving it and learned new things 🙌

jiegillet commented on concepts/mapsets/about.md:
````diff
@@ -0,0 +1,83 @@
+
+```elixir
+MapSet.new()
+# => MapSet.new([])
````
I think it used to be something like `#MapSet<>` or whatever, but they've been moving away from these notations to things that you can copy/paste. In that sense, this is the best way to represent sets :)

angelikatyborska replied:
That's so clever! I've always been annoyed at those `#` things that I couldn't copy-paste. I think they showed up most often for Ecto structs.

jiegillet commented on exercises/concept/gotta-snatch-em-all/.docs/instructions.md:
```diff
@@ -0,0 +1,100 @@
+You cannot trade a card you don't have, and you shouldn't trade a card for one that you already have.
+
+Implement `trade_card`, that takes two cards to trade (yours and theirs) and your current collection.
+The return value is a tuple of two values: a `Bool` stating if the trade is possible and worth doing, and the collection you would end up with if you did the trade (even if it's not actually possible).
```
Nice catch, it's a leftover from the Elm instructions, where `Bool` is a type.

jiegillet commented:
> very good `about.me`

Sorry, but this was not about you, it was about sets! 🤣 

I'm glad you like the PR, thanks for the review, I've addressed your comments in the relevantly named commits.

angelikatyborska reviewed (approved):
🚢
